### DISCUSSION TITLE
how to build observers to predict total concentration in peripheral venous blood in plasma

### DISCUSSION THREAD

**Discussion by jelllyyyyy** [Open-Systems-Pharmacology/Forum#1595] [Questions & Problems]
hello, I wonder how to build observer building block to predict total concentration of my 2 molcules of plasma concentration in peripheral venous blood..... 
I tried to build it, but don't know how to get the sum of 2 molecules.
also, after building the observer... no concentration profiles are shown in the modeling... so sad
please someone helppp me

**Comment by Yuri05** [comment]
See an example here: [SumOfConcentrations.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/12388325/SumOfConcentrations.zip)
Here, the sum of the concentration is defined in 2 different ways.
![grafik](https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/24181d82-313f-40b9-9383-3dc18d780c06)

Please note that the second option (which directly sums up the 2 observers) is easier to implement but less accurate
(when _observers_ are used in formulas, their value at the **previous time point** is taken, so in fact the second option calculates 
 $C_1\(t_{n-1}\)+C_2\(t_{n-1}\)$ instead of  $C_1\(t_n\)+C_2\(t_n\)$

![grafik](https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/5f0f4c0e-401c-44c1-b6fd-232ee4894725)

**Reply by jelllyyyyy** [reply]
I tried it but I can't see a graph after the simulation of it. I added an observer which is option 2 and run the simulation, but can't find the sum of concentration simualtion... do you know how to do that?

**Reply by jelllyyyyy** [reply]
i revised your file. I appreciate your help:)

**Reply by wilbertdew** [reply]
@jelllyyyyy Did you select it as an output after clicking "Define Settings and Run"?

**Reply by SCZ1174** [reply]
Hi, I have some questions when trying the above file.：
1. Which part of the content is the "MOLECULE" mapped to? From the result, it seems to be concound_1, but why? What exactly does "MOLECULE" represent and where can I find it. (I can't find it in "Possible referenced objects")
![SYG3ISFIJST %$VG 2AE6NY](https://github.com/Open-Systems-Pharmacology/Forum/assets/86552551/767e63ae-59bf-4bde-865d-7af40bf1844e)

**Reply by wilbertdew** [reply]
Hi @SCZ1174, these observers are in principle applied to all compounds that are present in the relevant compartment. "MOLECULE" in the path will be replaced by all the molecules that are in that compartment, and for each of those molecule, the observer will be calculated and you will be able to select the observer output for each molecule. I believe this is indeed not in the possible referenced object, but you can select any molecule in the right location to generate the path for that molecule and then replace the name of that molecule in the path by "MOLECULE" (you can just change the path and type this, the possible referenced objects is not mandatory to use)

**Reply by Yuri05** [reply]
As @wilbertdew said: if something (observer, passive transport, etc.) is created for **several molecules** in a simulation: the _path keyword_ `MOLECULE` is replaced by the name of a concrete molecule.
It's 1:N relationship between a building block and simulation.

![grafik](https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/1f5e26aa-7ba3-49bc-bbdc-ebcc93668b09)

In your particular case, it's 1:1 relationship, because the observer is defined only for the molecule **Compound_1**.
So here **Compound_1** could be used in the referenced paths instead of MOLECULE. As soon as it's not 1:1, you would need the keyword in any case, to avoid the duplication of observer definition.
![grafik](https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/7095bb00-6217-4286-89af-a79d392a125f)

s. also [Working with formulas](https://docs.open-systems-pharmacology.org/working-with-mobi/mobi-documentation/model-building-components#working-with-formulas) in the OSP documentation.
